tests/test_plugins.py

```diff
@@ -13,3 +14,33 @@ def test_runtime_context_available():
     for plugin in plugins:
         plugin.initialize()
         assert hasattr(plugin, 'runtime_context')
+
+
+@patch('aws_xray_sdk.core.plugins.ec2_plugin.do_request')
+def test_ec2_plugin(mock_do_request):
```

Comment:
Can we turn this into 3 unit tests, one for each condition it checks? That way in the future if one test breaks it will be easier to know what specifically is broken.

Reply:
Good point

Reply:
Broke down into 3 tests